[Edit] C++ Math Functions: `cosh()` (#1069)
Codecademy/docs: content/cpp/concepts/math-functions/terms/cosh/cosh.md

  - 'paths/computer-science'
---

> The *cosh* function returns the hyperbolic cosine of an argument given in radians.

Reviewer: ```suggestion
The **`cosh()`** function returns the hyperbolic cosine of an argument given in radians.
```

## Syntax

```cpp
std::cosh(n)
```

> where `n` is the angle in `radians`

Reviewer: ```suggestion
Where `n` is the angle in `radians`.
```

Reviewer 2: @kitrak-rev We still need to capitalize this word.

Reviewer 2: Actually, could we consider this suggestion?

```suggestion
The `angle` is measured in radians.
```

> `Return type is double, float or long double.

Reviewer: ```suggestion
Return type is double, float or long double.
```

If the magnitude of the result is too large in terms of the return type, the function returns HUGE_VAL with the proper sign, and an overflow range error occurs.

## Mathematic Formula:

> cosh(x) =  (e <sup>x</sup> + e<sup>−x</sup> )/2

Reviewer: ```suggestion
cosh(x) =  (e<sup>x</sup> + e<sup>−x</sup> )/2
```

## Example 1

> Use `cosh()` to return the hyperbolic cosine of `0.0` radians.

Reviewer: ```suggestion
This example uses `cosh()` to return the hyperbolic cosine of `0.0` radians:
```

  - 'paths/computer-science'
---

The **`cosh()`** function returns the hyperbolic cosine of an argument given in radians.

## Syntax

> ```cpp

Reviewer 2: ```suggestion
```pseudo
```

> std::cosh(n)

Reviewer 2: ```suggestion
std::cosh(angle)
```

Reviewer 2: I think `angle` better describes the parameter used for this function.

```

> The `angle` is measured in radians.
> The return type is a `double`, `float`, or `long double`.

Reviewer 2: ```suggestion
The `angle` is measured in radians and the return type is a `double`, `float`, or `long double`.
```

> Return type is double, float or long double.
> 
> If the magnitude of the result is too large in terms of the return type, the function returns HUGE_VAL with the proper sign, and an overflow range error occurs.

Reviewer 2: ```suggestion
The return type is a `double`, `float`, or `long double`.

If the magnitude of the result is too large to express, the function returns `HUGE_VAL` (positive or negative) and an overflow range error is thrown.
```

## Mathematic Formula

> cosh(x) = (e<sup>x</sup> + e<sup>−x</sup> )/2

Reviewer 2: ```suggestion
cosh(x) = (e<sup>x</sup> + e<sup>−x</sup> )/2
```

Let's wrap this in a \```tex code snippet:

\```tex
cosh(x) = (e^x + e^-x )/2
\```

> ## Example 1

Reviewer 2: ```suggestion
## Example
```

  // Output: The hyperbolic cosine of 0 radians is 1
}
```

## Example 2

> Use `cosh()` to return the hyperbolic cosine of `1.0` radians.

Reviewer: ```suggestion
This example uses `cosh()` to return the hyperbolic cosine of `1.0` radians:
```

  // Output: The hyperbolic cosine of 0 radians is 1
}
```

## Example 2

This example uses `cosh()` to return the hyperbolic cosine of `1.0` radians:

```cpp
#include <iostream>
#include <cmath>

int main() {
  double x = 1.0;
  double result;

  result = std::cosh(x);

>   std::cout << "The hyperbolic cosine of " << x << " radians is " << result << "\n";
>   // Output: The hyperbolic cosine of 1.0 radians is 1.54308

Reviewer 2: ```suggestion
  std::cout << "The hyperbolic cosine of " << x << " radians is " << result << "\n";
```
